src/network.ts

```diff
@@ -0,0 +1,155 @@
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import { fs } from 'mz';
+import * as networkCheck from 'network-checker';
+import * as os from 'os';
+import * as url from 'url';
+
+import * as constants from './lib/constants';
+import { EEXIST } from './lib/errors';
+import { checkTruthy } from './lib/validation';
+
+import blink = require('./lib/blink');
+
+const networkPattern = {
+	blinks: 4,
+	pause: 1000,
+};
+
+let isConnectivityCheckPaused = false;
+let isConnectivityCheckEnabled = true;
+
+function checkHost(
+	opts: networkCheck.ConnectOptions,
+): boolean | PromiseLike<boolean> {
+	if (!isConnectivityCheckEnabled || isConnectivityCheckPaused) {
+		return true;
+	}
+	return networkCheck.checkHost(opts);
+}
+
+function customMonitor(
+	options: networkCheck.ConnectOptions,
+	fn: networkCheck.MonitorChangeFunction,
+) {
+	return networkCheck.monitor(checkHost, options, fn);
+}
+
+export function enableCheck(enable: boolean) {
+	isConnectivityCheckEnabled = enable;
+}
+
+async function vpnStatusInotifyCallback(): Promise<void> {
+	try {
+		await fs.lstat(constants.vpnStatusPath);
+		isConnectivityCheckPaused = true;
+	} catch {
+		isConnectivityCheckPaused = false;
+	}
+}
+
+export const startConnectivityCheck = _.once(
+	async (
+		apiEndpoint: string,
+		enable: boolean,
+		onChangeCallback?: networkCheck.MonitorChangeFunction,
+	) => {
+		enableConnectivityCheck(enable);
+		if (!Boolean(apiEndpoint)) {
```

Comment:
What is the purpose of `Boolean`?
Is there a case where it transforms the truthiness of something?